systemFlake.nix

```diff
@@ -66,13 +69,19 @@ let
       extraArgs = sharedExtraArgs // lhs.extraArgs // rhs.extraArgs;
     };
 
-  foldHosts = foldl'
-    (lhs: rhs:
-      let output = rhs.name; in
-      # manually merge whats inside the output to prevent // override
-      lhs // { ${output} = lhs.${output} or { } // rhs.value; }
-    )
-    { };
+  mergeAny = lhs: rhs:
+    lhs // mapAttrs
+      (name: value:
+        if isAttrs value then lhs.${name} or { } // value
+        else if isList value then lhs.${name} or [ ] ++ value
+        else value
+      )
+      rhs;
+
+  foldHosts = foldl' mergeAny { };
+
+  mergeArgs = lhs: rhs:
+    lhs // mapAttrs (name: value: if isAttrs value then mergeAny lhs.${name} or { } value else value) rhs;
```

Comment:
I was thinking of merging `otherArguments` and the rest automatically, by checking if things are an attrset. It would make the below code much simpler, no need to call mergeAny on each builder expression

Reply:
Does this merge lists, according to rhe merge semantics defined in previous `mergeHosts`?

Reply:
I think it's ok to use the mergeAny function consistently. It's a superset.